packages/webpack-config/webpack/webpack.common.js

```diff
@@ -158,6 +158,14 @@ module.exports = async function(env = {}, argv) {
   const { noJavaScriptMessage } = config.web.dangerous;
   const noJSComponent = createNoJSComponent(noJavaScriptMessage);
 
+  if (
+    (isProd && config.web.build.deepScopeAnalysis !== false) ||
+    (isDev && config.web.build.deepScopeAnalysis != null)
```

Comment:
So for `--dev` builds, deep scope analysis is enabled even if you configure `
{ expo: web: { build: { deepScopeAnalysis: false } } } }`? That seems confusing.